server.go

```diff
@@ -169,9 +232,58 @@ func (server *Server) pingRoute(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(response)
 }
 
+func (server *Server) versionRoute(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	response := VersionResponse{Version: WAGGLE_VERSION}
+	json.NewEncoder(w).Encode(response)
+}
+
+func (server *Server) signersHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+		server.signersRoute(w, r)
+		return
+	case http.MethodPost:
+		var signer string
+		routePathSlice := strings.Split(r.URL.Path, "/")
+		requestedSigner := common.HexToAddress(routePathSlice[2])
+		for s := range server.AvailableSigners {
```

Comment:
It would be better if `server.AvailableSigners` was a `map[common.Address]Signer` (whatever the value struct is, I don't know).

Then you could do:
s, ok := server.AvailableSigners[requestedSigner]
if !ok {
    http.Error(w, fmt.Sprintf("Unacceptable signer provided %s", signer), http.StatusBadRequest)
    return
}

// sign stuff

Reply:
And if wrong signer is provided, maybe return 404 instead of 400?